### PR TITLE
Fix timeout error

### PR DESCRIPTION
**Pull request by Joel-Cripps**
Fixes invocation failure notices that are triggered by a timeouts. 

Event(timer trigger) functions, must return a callback(null.sometext)

Http functions can return a response.setStatus(200/500) and then return the status as a buffer object. response.send("")

### CODE REVIEW

File: callHealthCheck.ts [hunk 1]
```diff
@@ -20,12 +20,13 @@ exports.callHealthCheck = async function(){
         };
         try {
             console.log(`Calling ${FULL_URL}`)
-            const response = await axios.get(url, options);
-            console.log(response.data)
-            return response.data;
+            await axios.get(url, options);
         } catch (err) {
-            throw console.error(`Error retrieving VIP data from Fortigate: ${url}. Error : ${err} `);
+            throw console.error(`Error in CallHealthCheck Function. Error : ${err} `);
```
Comment (jamie-pate) [inline callHealthCheck.ts:25]: console.error returns undefined, you can't throw it.

Comment (jamie-pate) [reply callHealthCheck.ts:25]: if you are using callbacks, you need to `callback(err)` here instead of throwing anyways.

File: index.ts [hunk 2]
```diff
@@ -382,6 +387,12 @@ exports.main = async (context, req, res): Promise<void> => {
                 }
             }
         }
+        //Response is a buffer, expects a string.
+        //Sending the response should start the termination of the function, which is the main goal.
+        response.setStatusCode(200)
+        response.setHeader('content-type', 'application/json')
+        response.send('')
```
Comment (jamie-pate) [inline index.ts:394]: request.end() doesn't work here? :)

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: index.ts (hunk 2 above)
Comment (Joel-Cripps) [reply index.ts:394]: From what I tested it works to end the function but still prints a red "invocation error" on the function, which is not ideal.

File: callHealthCheck.ts (hunk 1 above)
Comment (Joel-Cripps) [reply callHealthCheck.ts:25]: added callback(err) and a console log for alicloud logging.

File: callHealthCheck.ts [hunk 3]
```diff
@@ -1,32 +1,34 @@
-import axios from 'axios';
```
Comment (jamie-pate) [inline callHealthCheck.ts:1]: the switch from `'` to `"` seems backwards? The linter/prettify config should prefer `'`

File: callHealthCheck.ts [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
-        } catch (err) {
-            throw console.error(`Error retrieving VIP data from Fortigate: ${url}. Error : ${err} `);
-        }
-    }
+exports.callHealthCheck = async function(
+  event,
+  context,
+  callback
+): Promise<void> {
+  console.log(`Timer Function Called`);
```
Comment (jamie-pate) [inline callHealthCheck.ts:11]: should also prefer `'` here since there's no interpolation happening

File: callHealthCheck.ts [hunk 5]
```diff
@@ -1,32 +1,34 @@
...
+    callback(err);
+  }
+  // Terminate function.
+  callback(null, "CallHealthCheck Terminated.");
+};
 if (module === require.main) {
-    exports.callHeallthCheck(console.log);
-}
\ No newline at end of file
+  exports.callHeallthCheck(console.log);
```
Comment (jamie-pate) [inline callHealthCheck.ts:33]: hea**l**lthcheck ?

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: callHealthCheck.ts (hunk 3 above)
Comment (Joel-Cripps) [reply callHealthCheck.ts:1]: Yes, added in the .prettierrc file.

File: callHealthCheck.ts (hunk 4 above)
Comment (Joel-Cripps) [reply callHealthCheck.ts:11]: Fixed.

File: callHealthCheck.ts [hunk 6]
```diff
@@ -1,32 +1,30 @@
...
+        await axios.get(url, options);
+    } catch (err) {
+        console.error(`Error in CallHealthCheck Function. Error : ${err} `);
+        callback(err);
     }
+    // Terminate function.
+    callback(null, 'CallHealthCheck Terminated.');
+};
 if (module === require.main) {
     exports.callHeallthCheck(console.log);
```
Comment (jamie-pate) [inline callHealthCheck.ts:29]: Still hea**ll**thCheck, which is obviously wrong right?

**Review by jamie-pate — CHANGES_REQUESTED** [review]

File: callHealthCheck.ts (hunk 6 above)
Comment (Joel-Cripps) [reply callHealthCheck.ts:29]: This section is more for local testing, but it calls the health check function if that is what you are asking.

**Review by jamie-pate — APPROVED** [review]

**Pull request merged**